### ISSUE TITLE
NPM 3

### ISSUE THREAD

**Issue by WaKeMaTTa** [kunalgolani/eslint-config#2]
Check this warnings:

```console
npm WARN peerDependencies The peer dependency eslint@~3.2.0 included from eslint-config-react-native will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN peerDependencies The peer dependency eslint-config-esnext@~1.4.0 included from eslint-config-react-native will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN peerDependencies The peer dependency eslint-plugin-react@~5.2.2 included from eslint-config-react-native will no

npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN peerDependencies The peer dependency eslint-plugin-react-native@~1.1.0 included from eslint-config-react-native will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN peerDependencies The peer dependency babel-eslint@~6.1.0 included from eslint-config-esnext will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN peerDependencies The peer dependency eslint-plugin-babel@~3.3.0 included from eslint-config-esnext will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
eslint-plugin-react-native@1.1.0 node_modules/eslint-plugin-react-native
```

**Comment by kunalgolani** [comment]
with npm 3, two major things changed:
+ non-conflicting dependencies would all be installed flatly in the topmost `node_modules`
+ `peerDependencies` would not be automatically installed

`eslint-config` depends on `eslint` and some other packages that need to be installed at the parent `node_modules` level, i.e. at the same level as `eslint-config`, not inside it.

for npm < 3, this is achieved through `peerDependencies`, which are installed at the same level as the package specifying them. this throws warnings in npm >= 3.

for npm >=3 the dependencies are anyway installed flatly, so they would be at the same level as `eslint-config`

**Comment by kunalgolani** [comment]
the warnings are known and expected as a result of backward compatibility for npm < 3.

**Comment by WaKeMaTTa** [comment]
Thanks, for responding si quickly. :+1:
